File: packages/eqapi_types/lib/model/v1/telegram.dart
import 'package:eqapi_types/model/v1/v1_database.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'telegram.freezed.dart';
part 'telegram.g.dart';

@freezed
class TelegramV1 with _$TelegramV1  implements V1Database {
  const factory TelegramV1({
    required int id,
    required int eventId,
    required String type,
    required String schemaType,
    required String status,
    required String infoType,
    required DateTime pressTime,
    required DateTime reportTime,
    DateTime? validTime,
    int? serialNo,
    String? headline,
    required Map<String, dynamic> body,
  }) = _TelegramV1;

  factory TelegramV1.fromJson(Map<String, dynamic> json) =>
      _$TelegramV1FromJson(json);
}

File: api/database/src/db/schema/telegram.ts
import {
	bigint,
	integer,
	jsonb,
	pgTable,
	serial,
	text,
	timestamp,
} from "drizzle-orm/pg-core";

/** 電文 */
export const telegram = pgTable("telegram", {
	id: serial("id").primaryKey(),
	eventId: bigint("event_id", { mode: "bigint" }).notNull(),
	type: text("type").notNull(),
	schemaType: text("schema_type").notNull(),
	status: text("status").notNull(),
	infoType: text("info_type").notNull(),
	pressTime: timestamp("press_time", { withTimezone: true }).notNull(),
	reportTime: timestamp("report_time", { withTimezone: true }).notNull(),
	validTime: timestamp("valid_time", { withTimezone: true }),
	serialNo: integer("serial_no"),
	headline: text("headline"),
	body: jsonb("body").notNull(),
});
